lib/std/json.zig

```diff
@@ -1383,8 +1383,69 @@ pub const ParseOptions = struct {
         Error,
         UseLast,
     } = .Error,
+
+    /// If false, finding an unknown field returns an error.
+    ignore_unknown_fields: bool = false,
 };
 
+fn skipValue(tokens: *TokenStream) !void {
+    const ClosingType = enum { Array, Object };
+
+    const closing_depth = tokens.parser.stack;
+    var waiting_for_closing_type: ?ClosingType = null;
+
+    while (true) {
+        const token = (try tokens.next()) orelse return error.UnexpectedEndOfJson;
+        switch (token) {
+            .ArrayBegin, .ObjectBegin => {
+                // If it's already waiting for something to close, don't change
+                // it.
+                if (waiting_for_closing_type != null) continue;
+
+                waiting_for_closing_type = switch (token) {
+                    .ArrayBegin => .Array,
+                    .ObjectBegin => .Object,
+                    else => unreachable,
+                };
+            },
+            .ArrayEnd, .ObjectEnd => {
+                const currently_closing_type: ClosingType = switch (token) {
+                    .ArrayEnd => .Array,
+                    .ObjectEnd => .Object,
+                    else => unreachable,
+                };
+
+                // waiting_for_closing_type can be null if the JSON is messed
+                // up, so return an error.
+                if (waiting_for_closing_type == null) {
+                    return error.UnexpectedToken;
+                }
+
+                if (waiting_for_closing_type.? == currently_closing_type and
+                    tokens.parser.stack == closing_depth) return;
+            },
+
+            else => {
+                // We only need to step through the token for the rest to work,
+                // unless we're waiting for something to close.
+                if (waiting_for_closing_type == null) return;
+            },
+        }
+    }
+}
+
+test "skipValue" {
+    try skipValue(&TokenStream.init("false"));
+    try skipValue(&TokenStream.init("true"));
+    try skipValue(&TokenStream.init("null"));
+    try skipValue(&TokenStream.init("42"));
+    try skipValue(&TokenStream.init("42.0"));
+    try skipValue(&TokenStream.init("\"foo\""));
+    try skipValue(&TokenStream.init("[101, 111, 121]"));
+    try skipValue(&TokenStream.init("{}"));
+    try skipValue(&TokenStream.init("{\"foo\": \"bar\"}"));
```

Comment:
Could you add the other (failing) test cases from #5852?

Reply:
I had to increase the number of nestings for [one of them](https://github.com/ziglang/zig/pull/5852/commits/404bbd385592e79c939f851b9d17c94d209bdde0#diff-defc5ed2d34b24434f31f2d92e270f4f98a1a32ec07147799eb12c1242d5b0a4R1464) to fail, but I couldn't reproduce [the last one](https://github.com/ziglang/zig/pull/5852/commits/404bbd385592e79c939f851b9d17c94d209bdde0#diff-defc5ed2d34b24434f31f2d92e270f4f98a1a32ec07147799eb12c1242d5b0a4R1469). Could you give me some feedback on it?

Reply:
What happended when you tried that last case?

Reply:
I think you meant `tokens.parser.stack_used < original_depth`, since `tokens.parser.stack` is the stack itself, not its depth.

Reply:
Applied the corrections. This will also return an error on empty arrays, so I added a test for that. Is this behavior desirable?